src/k8s/api/v1/cluster_config.go

```diff
@@ -103,6 +107,21 @@ type MetricsServerConfig struct {
 
 func (c MetricsServerConfig) GetEnabled() bool { return getField(c.Enabled) }
 
+type UserFacingDatastoreConfig struct {
+	// Type of the datastore. Needs to be "external".
+	Type       *string   `json:"type,omitempty" yaml:"type,omitempty"`
+	Servers    *[]string `json:"servers,omitempty" yaml:"servers,omitempty"`
+	CACert     *string   `json:"ca-crt,omitempty" yaml:"ca-crt,omitempty"`
+	ClientCert *string   `json:"client-crt,omitempty" yaml:"client-crt,omitempty"`
+	ClientKey  *string   `json:"client-key,omitempty" yaml:"client-key,omitempty"`
+}
```

Comment:
Hm, I see your point. However, in the `BootstrapConfig` we can also set the `k8s-dqlite` port which is obviously a datastore-related config. However, we don't want to set this port during runtime (do we?). 
Hence, I think the flat key approach makes sense for the bootstrap config. 

But I'm also open for discussions here.

Reply:
One approach we can take for now is re-use the `UserFacingDatastoreConfig` in the `UpdateClusterConfigRequest`, which will simply reject fields that are not allowed to changed (e.g. the k8s-dqlite port).

This also gives us a path forward _if_ we ever want to implement this in the future (we probably do not)

Reply:
I'm not a big fan of this. The API would suggest something we don't support. _IF_ we want to support it later, it would probably be a breaking change, hence `v2` anyway, right?